pkg/metrics/certificates_test.go

```diff
@@ -247,42 +249,52 @@ func TestCertificateMetrics(t *testing.T) {
 	for n, test := range tests {
 		t.Run(n, func(t *testing.T) {
 			m := New(testr.New(t), clock.RealClock{})
-			m.UpdateCertificate(test.crt)
 
-			if err := testutil.CollectAndCompare(m.certificateNotAfterTimeSeconds,
+			fakeClient := fake.NewSimpleClientset()
+			factory := externalversions.NewSharedInformerFactory(fakeClient, 0)
+			certsInformer := factory.Certmanager().V1().Certificates()
+			err := certsInformer.Informer().GetIndexer().Add(test.crt)
```

Comment:
```suggestion

			err := certsInformer.Informer().GetIndexer().Add(test.crt)
```

Nit: For consistency with the other test.